### PR TITLE
Clean usage

### PR DESCRIPTION
**Pull request by kolmanv**
Now it is very easy to use our program:
ruby -Ilib bin/content_server --conf_file=examples/config_content_server.yml
or
ruby -Ilib bin/backup_server --conf_file=examples/config_backup_server.yml

After this merge I will build new gems and check all works in gems too.

### CODE REVIEW

File: bin/content_server [hunk 1]
```diff
@@ -5,25 +5,15 @@
...
-  print "2"
   require 'params'
-  print "3"
   require 'run_in_background'
-  print "4"
   require 'content_server'
-  print "5"
 rescue LoadError
-  print "6"
   require 'rubygems'
```
Comment (gshilin) [inline bin/content_server]: If you know, that 'rubygems' will help to load other gems, why wouldn't you load if first (maybe in begin...end), and then load others? Wouldn't it be simpler code?

File: examples/config_backup_server.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Content server parameters
```
Comment (gshilin) [inline examples/config_backup_server.yml:1]: It is generally bad idea to have configuration with hardcoded values. It is much better to keep an example of this file (even with real data) and to permit everyone to copy it and to change as he wants. Otherwise somebody will one day commit his changes to local setup and you'll propagate it to production. The same is true also for config_content_server.yml

File: lib/content_server.rb [hunk 3]
```diff
@@ -27,8 +25,6 @@ module ContentServer
   Params.integer('backup_file_listening_port', 4444, 'Listening port in backup server for files')
   Params.path('content_data_path', '~/.bbfs/var/content.data',
```
Comment (gshilin) [inline lib/content_server.rb]: I believe you wanted to use here (in this file, I mean) values from configuration files.

File: lib/params.rb [hunk 4]
```diff
@@ -119,6 +121,13 @@ def value_type_check(value)
                       "is:'#{value.class}'.")
             end
           end
+        when 'Complex' then
+          if not @value.nil?
```
Comment (gshilin) [inline lib/params.rb]: "if not" in plain English is "unless"

File: lib/params.rb [hunk 5]
```diff
@@ -185,6 +198,12 @@ def Params.override_param(name, value)
       existing_param.value = value.to_s
     elsif existing_param.type.eql?('Path')
       existing_param.value = File.expand_path(value.to_s)
+    elsif existing_param.type.eql?('Complex')
+      if value.class.eql?(Hash) || value.class.eql?(Array)
```
Comment (gshilin) [inline lib/params.rb:202]: Why are you expecting value to be really Hash? Have you ever heard of "Duck typing"?
There are different kinds of hashes, some of them support order, some enable keys to be string or symbol, but they are still hashes.
The same regarding Arrays. Actually File behaves exactly like Array, so why would you disable it?

File: lib/params.rb [hunk 6]
```diff
@@ -253,18 +279,26 @@ def Params.init(args)
...
 
     # Prints help and parameters if needed.
-    puts @help_messages unless @help_messages.empty?
+    if @show_help_and_exit
+      # Print parameters + description and exit.
+      puts "Full list of parameters:"
+      Params.each do |name, param|
+        puts "--#{name}, #{param.type}, default:#{param.value}\n\t#{param.desc}"
+      end
+      exit
```
Comment (gshilin) [inline lib/params.rb:288]: That's good idea to return value upon exit, they you'll be able to check exit status from external shell.

File: lib/params.rb [hunk 7]
```diff
@@ -303,7 +337,7 @@ def Params.parse_command_line_arguments(args)
       # Define help command for available options
       # executing --help will printout all pre-defined switch options
       opts.on_tail("-h", "--help", "Show this message") do
-        @help_messages << opts
+        @show_help_and_exit = true
```
Comment (gshilin) [inline lib/params.rb:340]: In OOP this is usually done via exceptions

File: bin/content_server (hunk 1 above)
Comment (kolmanv) [inline bin/content_server]: Removed the rescue.

File: examples/config_backup_server.yml (hunk 2 above)
Comment (kolmanv) [inline examples/config_backup_server.yml:1]: This is an example file, it is under the example directory.
It is not being used in production.

File: lib/params.rb (hunk 4 above)
Comment (kolmanv) [inline lib/params.rb]: Done.

File: lib/params.rb (hunk 5 above)
Comment (kolmanv) [inline lib/params.rb:202]: Is there a way to check that the class implements Hash or Array interfaces (via duck typing).

File: lib/params.rb (hunk 6 above)
Comment (kolmanv) [inline lib/params.rb:288]: The default is correct here (true).

File: lib/params.rb (hunk 7 above)
Comment (kolmanv) [inline lib/params.rb:340]: Don't follow, how?

File: lib/params.rb (hunk 5 above)
Comment (gshilin) [inline lib/params.rb:202]: value.respond_to?(:[])

File: lib/params.rb (hunk 7 above)
Comment (gshilin) [inline lib/params.rb:340]:
``` ruby
begin
  foo
rescue ShowHelpOnlyError
  show_help
  exit
end
```
